kaldb/src/main/java/com/slack/kaldb/metadata/core/KaldbMetadataStore.java

```diff
@@ -23,18 +26,31 @@
 import org.apache.curator.x.async.modeled.cached.ModeledCacheListener;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.data.Stat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * KaldbMetadataStore is a class which provides consistent ZK apis for all the metadata store class.
  *
  * <p>Every method provides an async and a sync API. In general, use the async API you are
  * performing batch operations and a sync if you are performing a synchronous operation on a node.
+ *
+ * <p><a href="https://curator.apache.org/docs/recipes-persistent-node">Persistent node recipie</a>
  */
 public class KaldbMetadataStore<T extends KaldbMetadata> implements Closeable {
+  private static final Logger LOG = LoggerFactory.getLogger(KaldbMetadataStore.class);
+
+  public static String PERSISTENT_EPHEMERAL_PROPERTY = "kaldb.metadata.persistentEphemeral";
```

Comment:
It seems that `Persistent Node` is a lot of complexity esp when we can use a non-ephermeral node add for cleaner tests. Can we use a regular node instead since this is mostly needed for HPA?